Drupal 9 readiness

Fix deprecation warnings for Drupal 9, and mark the module as compatible by adding `core_version_requirement: ^8 || ^9` to all module info files (https://www.drupal.org/node/3070687).  (https://www.drupal.org/node/3070687). 
The following report was generated using [mglaman/drupal-check](https://github.com/mglaman/drupal-check):

```
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   apigee_devportal_kickstart.install                                                                                                                                            
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  233    Call to deprecated function file_default_scheme():                                                                                                                            
         in drupal:8.8.0 and is removed from drupal:9.0.0. Use                                                                                                                         
         \Drupal::config('system.file')->get('default_scheme') instead.                                                                                                                
  265    Call to deprecated constant FILE_CREATE_DIRECTORY: Deprecated in drupal:8.7.0 and is removed from drupal:9.0.0. Use \Drupal\Core\File\FileSystemInterface::CREATE_DIRECTORY.  
  265    Call to deprecated function file_prepare_directory():                                                                                                                         
         in drupal:8.7.0 and is removed from drupal:9.0.0.                                                                                                                             
         Use \Drupal\Core\File\FileSystemInterface::prepareDirectory().                                                                                                                
  267    Call to deprecated function file_unmanaged_copy():                                                                                                                            
         in drupal:8.7.0 and is removed from drupal:9.0.0.                                                                                                                             
         Use \Drupal\Core\File\FileSystemInterface::copy().                                                                                                                            
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ ------------------------------------------------------------------------------------------------ 
  Line   modules/custom/apigee_kickstart_enhancement/apigee_kickstart_enhancement.module                 
 ------ ------------------------------------------------------------------------------------------------ 
  285    Call to deprecated method getLowercaseLabel() of class Drupal\Core\Entity\EntityTypeInterface:  
         in drupal:8.8.0 and is removed from drupal:9.0.0.                                               
         Instead, you should call getSingularLabel().                                                    
         See https://www.drupal.org/node/3075567                                                         
  299    Call to deprecated method getLowercaseLabel() of class Drupal\Core\Entity\EntityTypeInterface:  
         in drupal:8.8.0 and is removed from drupal:9.0.0.                                               
         Instead, you should call getSingularLabel().                                                    
         See https://www.drupal.org/node/3075567                                                         
  306    Call to deprecated method getLowercaseLabel() of class Drupal\Core\Entity\EntityTypeInterface:  
         in drupal:8.8.0 and is removed from drupal:9.0.0.                                               
         Instead, you should call getSingularLabel().                                                    
         See https://www.drupal.org/node/3075567                                                         
 ------ ------------------------------------------------------------------------------------------------ 

 ------ -------------------------------------------------------------------------- 
  Line   themes/custom/apigee_kickstart/apigee_kickstart.theme                     
 ------ -------------------------------------------------------------------------- 
  28     Call to deprecated function entity_load_multiple_by_properties():         
         in drupal:8.0.0 and is removed from drupal:9.0.0. Use the                 
         entity type storage's loadByProperties() method.                          
  35     Call to deprecated function entity_view():                                
         in drupal:8.0.0 and is removed from drupal:9.0.0.                         
         Use the entity view builder's view() method for creating a render array:  
 ------ -------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 9 errors 
```

--- a/apigee_devportal_kickstart.info.yml
+++ b/apigee_devportal_kickstart.info.yml
@@ -1,7 +1,7 @@
 name: Apigee Developer Portal Kickstart
 type: profile
 description: 'A fast demo and starting point for Apigee Developer Portals.'
-core: 8.x
+core_version_requirement: ^8 || ^9
 
 # Optional: Declare your installation profile as a distribution
 # This will make the installer auto-select this installation profile.

--- a/apigee_devportal_kickstart.install
+++ b/apigee_devportal_kickstart.install
@@ -1,5 +1,7 @@
 <?php
 
+use Drupal\Core\File\Exception\FileException;
+use Drupal\Core\File\FileSystemInterface;
 /**
  * @file
  * Copyright 2019 Google Inc.
@@ -230,7 +232,7 @@ function apigee_devportal_kickstart_install() {
   $src_path = drupal_get_path('profile', 'apigee_devportal_kickstart') . '/resources/files';
 
   // Get public files directory.
-  $dest_path = file_default_scheme() . '://';
+  $dest_path = \Drupal::config('system.file')->get('default_scheme') . '://';
 
   // Files to copy.
   // Not all files are copied. So we list them here individually.
@@ -258,13 +260,20 @@ function apigee_devportal_kickstart_install() {
   foreach ($filesets as $destination => $files) {
     foreach ($files as $file) {
       if (file_exists($src_path . '/' . $file)) {
+        $file_system = \Drupal::service('file_system');
+
         // If the file exists in public://, it should be skipped, as these are
         // tied to media entities which hard-code the path/file.
         $directory = $dest_path . $destination;
         if (!empty($destination)) {
-          file_prepare_directory($directory, FILE_CREATE_DIRECTORY);
+          $file_system->prepareDirectory($directory, FileSystemInterface::CREATE_DIRECTORY);
         }
-        file_unmanaged_copy($src_path . '/' . $file, $directory . $file);
+        try {
+          \Drupal::service('file_system')->copy($src_path . '/' . $file, $directory . $file);
+        }
+        catch (FileException $e) {
+          // Do nothing.
+        }
       }
     }
   }

--- a/modules/custom/apigee_kickstart_content/apigee_kickstart_content.info.yml
+++ b/modules/custom/apigee_kickstart_content/apigee_kickstart_content.info.yml
@@ -1,7 +1,7 @@
 name: Apigee Kickstart Content
 description: "Provides demo content for Apigee Developer Portal Kickstart."
 type: module
-core: 8.x
+core_version_requirement: ^8 || ^9
 package: Apigee Kickstart
 dependencies:
   - default_content:default_content

--- a/modules/custom/apigee_kickstart_customizer/apigee_kickstart_customizer.info.yml
+++ b/modules/custom/apigee_kickstart_customizer/apigee_kickstart_customizer.info.yml
@@ -1,5 +1,5 @@
 name: Apigee Kickstart Customizer
 description: "Provides a UI for customizing the Apigee Kickstart theme."
 type: module
-core: 8.x
+core_version_requirement: ^8 || ^9
 package: Apigee Kickstart

--- a/modules/custom/apigee_kickstart_enhancement/apigee_kickstart_enhancement.info.yml
+++ b/modules/custom/apigee_kickstart_enhancement/apigee_kickstart_enhancement.info.yml
@@ -1,7 +1,7 @@
 name: Apigee Kickstart Enhancement
 description: "Provides enhancement for Apigee Edge module."
 type: module
-core: 8.x
+core_version_requirement: ^8 || ^9
 package: Apigee Kickstart
 dependencies:
   - apigee_edge:apigee_edge

--- a/modules/custom/apigee_kickstart_enhancement/apigee_kickstart_enhancement.module
+++ b/modules/custom/apigee_kickstart_enhancement/apigee_kickstart_enhancement.module
@@ -282,7 +282,7 @@ function _apigee_kickstart_enhancement_check_app_warnings(AppInterface $app, Ent
   foreach ($app->getCredentials() as $credential) {
     if ($credential->getStatus() === AppCredential::STATUS_REVOKED) {
       $args = [
-        '@app' => $entity_type->getLowercaseLabel(),
+        '@app' => mb_strtolower($entity_type->getSingularLabel()),
       ];
       if (count($app->getCredentials()) > 1) {
         $warnings['revokedCred'] = t('One of the credentials associated with this @app is in revoked status.', $args);
@@ -296,17 +296,17 @@ function _apigee_kickstart_enhancement_check_app_warnings(AppInterface $app, Ent
     // Check for expired credentials.
     if (($expired_date = $credential->getExpiresAt()) && Drupal::time()->getRequestTime() - $expired_date->getTimestamp() > 0) {
       $warnings['expiredCred'] = t('At least one of the credentials associated with this @app is expired.', [
-        '@app' => $entity_type->getLowercaseLabel(),
+        '@app' => mb_strtolower($entity_type->getSingularLabel()),
       ]);
     }
 
     foreach ($credential->getApiProducts() as $cred_product) {
       if ($cred_product->getStatus() == CredentialProduct::STATUS_REVOKED || $cred_product->getStatus() == CredentialProduct::STATUS_PENDING) {
         $args = [
-          '@app' => $entity_type->getLowercaseLabel(),
-          '@api_product' => Drupal::entityTypeManager()
+          '@app' => mb_strtolower($entity_type->getSingularLabel()),
+          '@api_product' => mb_strtolower(Drupal::entityTypeManager()
             ->getDefinition('api_product')
-            ->getLowercaseLabel(),
+            ->getSingularLabel()),
           '@status' => $cred_product->getStatus() == CredentialProduct::STATUS_REVOKED ? t('revoked') : t('pending'),
         ];
         if (count($app->getCredentials()) === 1) {

--- a/modules/custom/apigee_kickstart_migrate/apigee_kickstart_migrate.info.yml
+++ b/modules/custom/apigee_kickstart_migrate/apigee_kickstart_migrate.info.yml
@@ -1,7 +1,7 @@
 name: Apigee Kickstart Migrate
 description: "Provides a migration path from Drupal 7 Devportal to Drupal 8 Kickstart."
 type: module
-core: 8.x
+core_version_requirement: ^8 || ^9
 package: Apigee Kickstart
 dependencies:
   - drupal:migrate

--- a/themes/custom/apigee_kickstart/apigee_kickstart.theme
+++ b/themes/custom/apigee_kickstart/apigee_kickstart.theme
@@ -25,14 +25,18 @@ function apigee_kickstart_add_regions_to_node(array $regions, &$variables) {
   $regions = array_intersect(array_keys($available_regions), $regions);
 
   foreach ($regions as $key => $region) {
-    $blocks = entity_load_multiple_by_properties('block', ['theme' => $theme, 'region' => $region]);
+    $blocks = \Drupal::entityTypeManager()
+      ->getStorage('block')
+      ->loadByProperties(['theme' => $theme, 'region' => $region]);
     uasort($blocks, 'Drupal\block\Entity\Block::sort');
 
     // Capture viewable blocks and their settings to $build.
     $build = [];
     foreach ($blocks as $key => $block) {
       if ($block->access('view')) {
-        $build[$key] = entity_view($block, 'block');
+        $build[$key] = \Drupal::entityTypeManager()
+          ->getViewBuilder($block->getEntityTypeId())
+          ->view($block, 'block');
       }
     }
 

--- a/themes/custom/apigee_kickstart/includes/form.inc
+++ b/themes/custom/apigee_kickstart/includes/form.inc
@@ -130,4 +130,3 @@ function apigee_kickstart_form_alter(array &$form, FormStateInterface $form_stat
     }
   }
 }
-

